packages/flutter_tools/lib/src/base/features.dart

```diff
@@ -0,0 +1,245 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../version.dart';
+import 'config.dart';
+import 'context.dart';
+import 'platform.dart';
+
+/// The current [FeatureFlags] implementation.
+///
+/// If not injected, a default implementation is provided.
+FeatureFlags get featureFlags => context.get<FeatureFlags>() ?? const FeatureFlags();
+
+/// The interface used to determine if a particular [Feature] is enabled.
+///
+/// The rest of the tools code should use this class instead of looking up
+/// features directly. To faciliate rolls to google3 and other clients, all
+/// flags should be provided with a default implementation here. Clients that
+/// use this class should extent instead of implement, so that new flags are
+/// picked up automatically.
+class FeatureFlags {
+  const FeatureFlags();
+
+  /// Whether flutter web is enabled.
+  ///
+  /// Defaults to `false`.
+  bool get isWebEnabled => false;
+
+  /// Whether flutter desktop for macOS is enabled.
+  ///
+  /// Defaults to `false`.
+  bool get isMacOSEnabled => false;
+
+  /// Whether flutter desktop for Windows is enabled.
+  ///
+  /// Defaults to `false`.
+  bool get isWindowsEnabled => false;
+
+  /// Whether flutter desktop for linux is enabled.
+  ///
+  /// Defaults to `false`.
+  bool get isLinuxEnabled => false;
+}
+
+/// All current Flutter feature flags.
+const List<Feature> allFeatures = <Feature>[
+  flutterWebFeature,
+  flutterLinuxDesktopFeature,
+  flutterMacOSDesktopFeature,
+  flutterWindowsDesktopFeature,
+];
+
+/// The [Feature] for flutter web.
+const Feature flutterWebFeature = Feature(
+  name: 'Flutter Web',
+  configSetting: 'flutter-web',
+  environmentOverride: 'FLUTTER_WEB',
```

Comment:
nit/minor: `ENABLE_FLUTTER_WEB`

Reply:
Similar to https://github.com/flutter/flutter/pull/36138#discussion_r303604941, this environment variable matches the old instructions on how to enable web.